pss/pss_test.go

```diff
@@ -495,6 +495,133 @@ func TestAddressMatchProx(t *testing.T) {
 	}
 }
 
+func TestMessageEnqueued(t *testing.T) {
+
+	// setup
+	privkey, err := crypto.GenerateKey()
+	if err != nil {
+		t.Fatal(err.Error())
+	}
+
+	addr := make([]byte, 32)
+	addr[0] = 0x01
+	ps := newTestPssStart(privkey, network.NewKademlia(addr, network.NewKadParams()), NewParams(), false)
+	outboxCapacity := 2
+	ps.outbox = make(chan *outboxMsg, outboxCapacity)
+
+	dequeue := func() *PssMsg {
+		select {
+		case oMsg := <-ps.outbox:
+			return oMsg.msg
+		default:
+			return nil
+		}
+	}
+
+	messageAddr := addr
+	topic := [4]byte{}
+	data := []byte{0x66, 0x6f, 0x6f}
+
+	msg := testMessage(messageAddr, topic, data)
+
+	err = ps.enqueue(msg, false)
+	if err != nil && ps.forwardPending != 1 {
+		t.Fatal("Message should have space for enqueuing")
+	}
+	t.Log("Message enqueued", "Outbox len", len(ps.outbox))
+
+	oMsg := dequeue()
+	if oMsg == nil {
+		t.Fatal("Message should have been dequeued")
+	}
+	if ps.forwardPending != 1 {
+		t.Fatal("Pending message should still be 1")
```

Comment:
Should _now_  - not _still_ - be `1`

Reply:
Refer to my previous comment. `forwardPending` is increased when a new message is enqueued in the system, even before the message is processed.